includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -248,8 +249,22 @@ public function recurring_checkout_session( $payload ) {
 					'currencyCode' => get_woocommerce_currency(),
 				);
 			}
-		} elseif ( $change_payment_for_subscription ) {
-			$subscription = wcs_get_subscription( absint( $_GET['change_payment_method'] ) );
+		} elseif ( $cart_contains_renewal || $change_payment_for_subscription ) {
+			if ( $cart_contains_renewal ) {
+				if ( ! isset( $cart_contains_renewal['subscription_renewal'] ) || ! isset( $cart_contains_renewal['subscription_renewal']['subscription_id'] ) ) {
+					return $payload;
+				}
+
+				$subscription = wcs_get_subscription( $cart_contains_renewal['subscription_renewal']['subscription_id'] );
+			} elseif ( $change_payment_for_subscription ) {
+				if ( ! isset( $_GET['change_payment_method'] ) ) {
+					return $payload;
+				}
+
+				$subscription = wcs_get_subscription( absint( $_GET['change_payment_method'] ) );
+			} else {
+				return $payload;
```

Comment:
This is just a failsafe mechanism, in case at some point i add something at the top. and don't specifically handle it here, I do nothing in that case.

Extra safety coding if you will. But totally true that it's not necessary.